### PR TITLE
Signin with GitHub

### PR DESCRIPTION
**Pull request by oudeismetis**

### CODE REVIEW

File: franklin/config/urls.py [hunk 1]
```diff
@@ -18,13 +18,17 @@
...
 
-from github.views import auth, callback, deploy_hook, register_repo
+from github.views import deploy_hook, register_repo
+from users.views import UserLogin
 
 urlpatterns = [
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^auth/', auth, name='auth'),
-    url(r'^callback/', callback, name='callback'),
+    url(r'^$', UserLogin.as_view()),
```
Comment (oudeismetis) [inline franklin/config/urls.py]: @JulianGindi Need to talk about the auth and callback methods in github views. Not sure if they are relevant anymore.

File: franklin/github/views.py [hunk 2]
```diff
@@ -92,39 +94,41 @@ def make_rest_post_call(url, headers, body):
...
+            # If the header has a paging link called 'next', update our url
+            # and continue with the while loop
+            if response.links and response.links.get('next', None):
+                url = response.links['next']['url']
+                have_next_page = True
+
+    if not repos:
+        logger.error('Failed to find repos for owner', owner_name)
+    return repos
+"""
```
Comment (oudeismetis) [inline franklin/github/views.py]: Just changing the commenting style per Julian's request

File: franklin/github/views.py [hunk 3]
```diff
@@ -172,6 +176,17 @@ def register_repo(request):
...
+        user = User.objects.get(...)
+        social = user.social_auth.get(provider='google-oauth2')
+        response = requests.get(
+                    'https://www.googleapis.com/plus/v1/people/me/people/visible',
+                        params={'access_token':
+                            social.extra_data['access_token']}
+                        )
+        friends = response.json()['items']
+        """
         serializer = SiteSerializer(data=request.data)
```
Comment (oudeismetis) [inline franklin/github/views.py]: Step 2 - list all of the users repos on the UI with a button called `activate` next to each.
Step 3 - clicking that button calls this endpoint for the user
Step 4 - We call github as the user and generate a `deploy_key` for the project being registered. 
Step 5 - $$

**Comment by JulianGindi** [comment]
Looks good to me. Tested locally? If so, let's get it in and test on the integration environment at some point tomorrow. :+1:

**Comment by oudeismetis** [comment]
yup, it's been tested and works. Will just need to create a github app for the integration environment

**Comment by oudeismetis** [comment]
merged

**Pull request closed without merging**